lib/pf/SwitchFactory.pm

```diff
@@ -28,9 +28,23 @@ use Benchmark qw(:all);
 use List::Util qw(first);
 use pf::ConfigStore::Switch;
 use pf::CHI;
+use pf::Switch;
+
+#Loading all the switch modules ahead of time
+use Module::Pluggable
+  search_path => [qw(pf::Switch)],
+  'require' => 1,
+  sub_name    => 'modules';
+
+our @MODULES = __PACKAGE__->modules;
+
+our %TYPE_TO_MODULE = map { my $type = $_; $type =~ s/^pf::Switch:://; $type => $_ } grep { $_->can('description') } @MODULES;
```

Comment:
I see you added the builder but this is stll here.

Remove the assignment and just declare it ?